weave/index/index_pandas.py

```diff
@@ -200,17 +204,21 @@ def generate_index(self, **kwargs):
 
     def _upload_index(self, index):
         """Upload a new index."""
-        with tempfile.TemporaryDirectory() as out:
-            n_secs = time_ns()
-            temp_json_path = os.path.join(out, f"{n_secs}-index.json")
-            index.to_json(temp_json_path, date_format="iso", date_unit="ns")
-            UploadBasket(
-                upload_items=[{"path":temp_json_path, "stub":False}],
-                basket_type=self.index_basket_dir_name,
-                file_system=self.file_system,
-                source_file_system=LocalFileSystem(),
-                pantry_path=self.pantry_path
-            )
+        n_secs = time_ns()
+        # If the pantry is read-only, don't upload the index.
+        if not self.pantry_read_only:
+            with tempfile.TemporaryDirectory() as out:
+                temp_json_path = os.path.join(out, f"{n_secs}-index.json")
+                index.to_json(temp_json_path,
+                              date_format="iso",
+                              date_unit="ns")
+                UploadBasket(
+                    upload_items=[{"path":temp_json_path, "stub":False}],
+                    basket_type=self.index_basket_dir_name,
+                    file_system=self.file_system,
+                    source_file_system=LocalFileSystem(),
+                    pantry_path=self.pantry_path
+                )
```

Comment:
Would it be worth raising an error when the user attempts to make a change to a read-only pantry. Perhaps, something like the following could be done.

```python
    else:
        raise RuntimeError("Attempting to write to a read-only panty.")  # Or use ValueError if that is the standard we're following.
```

Reply:
We do have this, but when we upload a basket. There is no need to here, because this is dealing with the index. We normally upload the index to the file system to keep it, but that is happening in the background. When we create an index, it is saved locally so the user can use it and interact with it, it can't be uploaded for future use in a read-only pantry, so we just leave it to be saved locally.

Reply:
Got it.